pymongo/topology_description.py

```diff
@@ -324,6 +324,16 @@ def apply_selector(
             description = self.server_descriptions().get(address)
             return [description] if description else []
 
+        # Primary selection fast path.
+        if self.topology_type == TOPOLOGY_TYPE.ReplicaSetWithPrimary and isinstance(
+            selector, Primary
```

Comment:
My one concern is that technically this could be a breaking change if a user is subclassing Primary and overriding `__call__` to do something special during server selection. One simple way to avoid that risk is doing `type(selector) is Primary` rather than isinstance. Or we could document the potential breaking change.

Reply:
I went with the code modification option